docs/animation/armature.md

```diff
@@ -0,0 +1,147 @@
+Armature Files
+==============
+
+Armature Files define joints and clips for animating a model.
+
+File Structure
+----------------
+
+An example armature file, taken from the forge debug mod (requires #3875)
+```json
+{
+  "joints": {
+    "stick": {"2": [1.0]},
+    "cube": {"3": [1.0]}
+  },
+  "clips": {
+    "default": {
+      "loop": true,
+      "joint_clips": {
+        "stick": [
+          {
+            "variable": "offset_x",
+            "type": "uniform",
+            "interpolation": "linear",
+            "samples": [0, 0.6875, 0]
+          }
+        ],
+        "cube": [
+          {
+            "variable": "offset_x",
+            "type": "uniform",
+            "interpolation": "linear",
+            "samples": [0, 0.6875, 0]
+          },
+          {
+            "variable": "axis_z",
+            "type": "uniform",
+            "interpolation": "nearest",
+            "samples": [ 1 ]
+          },
+          {
+            "variable": "origin_x",
+            "type": "uniform",
+            "interpolation": "nearest",
+            "samples": [ 0.15625 ]
+          },
+          {
+            "variable": "origin_y",
+            "type": "uniform",
+            "interpolation": "nearest",
+            "samples": [ 0.40625 ]
+          },
+          {
+            "variable": "angle",
+            "type": "uniform",
+            "interpolation": "linear",
+            "samples": [0, 120, 240, 0, 120, 240]
+          }
+        ]
+      },
+      "events": {}
+    }
+  }
+}
+
+```
+
+The file is organized in two sections, joints and clips.
+
+Joints
+--------
+Each joint defines a connection to the model file
+
+The format is like this:
+```json
+ "joints": {
+  "joint_name": {"index in model file": [1.0]}
+ }
+```
+
+- `joint_name` is the name of the joint
+- `index in model file` is a 0-indexed number denoting which model element this joint controls
+
+Not all elements need to have a joint, only the ones you are animating.
+
```

Comment:
fry said it was unused, something about blending the joint clips/TRSR's together